llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -1921,6 +1921,7 @@ MemoryDepChecker::getDependenceDistanceStrideAndSize(
   if (StrideAPtr && *StrideAPtr < 0) {
     std::swap(Src, Sink);
     std::swap(AInst, BInst);
+    std::swap(ATy, BTy);
```

Comment:
I think the only way this could currently go wrong is in the code around line 1939 that gets start and end for the Src & Sink? Would it be possible to add a test case for it? (The other uses should be fine below as we only check if they match

Reply:
Good catch: fixed now :)

Reply:
Thanks for adding the test, it shows an improvement instead of fix of miscompile, but that's great!